client.go

```diff
@@ -234,7 +236,15 @@ func (c *Client) ExecCmd(cmd *Cmd) ([]string, error) {
 		return nil, ErrNotConnected
 	}
 
-	c.work <- cmd.String()
```

Comment:
Channels are concurrent safe so why the mutex?

Reply:
If you execute the test `TestConcurrency` that's added in the commit you'll see that it wasn't concurrent safe at all (even before my pr - see the referenced issue).

If you send multiple commands very shortly after one another that return a long response from the query they cause a data-race. Only one command can be written at the same time due to the channel being concurrency safe but after the first command is written, another command can be executed even before a response is received which results in both responses being read by the first call and the second call timing out because the response it's waiting for has already been read.